### PR TITLE
Add a simple `aot.export` entrypoint for one-short nn.Module export.

### PR DESCRIPTION
**Pull request by stellaraccident**
* Adds full compilation support (as opposed to just stopping at the exported IR).
* Adds a sample which exercises the entire flow.

### CODE REVIEW

**Review by dan-garvey — APPROVED** [review]
looks great

**Comment by brucekimrokcmu** [comment]
LGTM!

File: tests/examples/aot_mlp_test.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import logging
+from pathlib import Path
+import sys
+import subprocess
+import unittest
+
+REPO_DIR = Path(__file__).resolve().parent.parent.parent
```
Comment (dan-garvey) [inline tests/examples/aot_mlp_test.py:13]: I kind of hate this but don't know a better way

**Review by dan-garvey — APPROVED** [review]

File: tests/examples/aot_mlp_test.py (hunk 1 above)
Comment (stellaraccident) [reply tests/examples/aot_mlp_test.py:13]: Yeah.

**Pull request merged**